core/tests/test-dependencies/test-to-modules-matcher.ts

```diff
@@ -0,0 +1,228 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview This script is used to collect all the Angular modules that are
+ * used by tests, by matching their URLs with Angular modules at runtime.
+ */
+
+import fs from 'fs';
+import path from 'path';
+import {
+  DefaultUrlSerializer,
+  UrlSegment,
+  UrlSegmentGroup,
+  Route,
+} from '@angular/router';
+import {Browser, Frame, Target} from 'puppeteer';
+import {getRouteToModuleMapping} from './route-to-module-mapping-generator';
+import {glob} from 'glob';
+
+// A mapping of common modules that many tests use, so by default we will exclude them
+// from being collected. Each excluded module also has an array of globs where if specified
+// any golden file under that glob will not be excluded from collecting the module.
+const COMMON_MODULES_TO_EXCLUDE: Record<string, string[]> = {
```

Comment:
How do we decide which files add to this constant?

Reply:
Regarding this, since the acceptance tests are being done by @Akhilesh-max, I took a look at his proposal and put the CUJs that would capture those modules. The modules are just modules that are common and used by every test (I believe these are the main ones and won't change much).

Reply:
Hmm, do not assume that the codebase will not change. For example what if someone decides to add a new splash page alogside the existing one (lets say a landing page with specific purpose), what do they need to do then in regards to this constant?

Reply:
Well if its a new page (new module not the main splash page) and a different URL, then most tests won't visit it and therefore doesn't need to be excluded. However, if it is visited by every page, then every test will alert that this module will need to be added and therefore the developer should instead look to add it to the constant.

Reply:
Okay, can we maybe notify the user somehow when there are too many tests that are alerting? If not it should at least be in the wiki.

Reply:
@vojtechjelinek Yes I will put that in the WIKI, quite hard to check if many tests are alerting since its in the CI.